modules/images/image-loading-optimization/storage/post-type.php

```diff
@@ -99,49 +99,42 @@ function ilo_parse_stored_url_metrics( WP_Post $post ): array {
 				json_last_error_msg()
 			)
 		);
-		$url_metrics = array();
-	} elseif ( ! is_array( $url_metrics ) ) {
+		$url_metrics_data = array();
+	} elseif ( ! is_array( $url_metrics_data ) ) {
 		$trigger_error(
 			sprintf(
 				/* translators: %s is post type slug */
 				__( 'Contents of %s post type was not a JSON array.', 'performance-lab' ),
 				ILO_URL_METRICS_POST_TYPE
 			)
 		);
-		$url_metrics = array();
+		$url_metrics_data = array();
 	}
 
 	return array_values(
 		array_filter(
-			$url_metrics,
-			static function ( $url_metric ) use ( $trigger_error ) {
-				// TODO: If we wanted, we could use the JSON Schema to validate the stored metrics.
-				$is_valid = (
-					is_array( $url_metric )
-					&&
-					isset(
-						$url_metric['viewport']['width'],
-						$url_metric['viewport']['height'],
-						$url_metric['elements']
-					)
-					&&
-					is_int( $url_metric['viewport']['width'] )
-					&&
-					is_array( $url_metric['elements'] )
-				);
-
-				if ( ! $is_valid ) {
-					$trigger_error(
-						sprintf(
-							/* translators: %s is post type slug */
-							__( 'Unexpected shape to JSON array in post_content of %s post type.', 'performance-lab' ),
-							ILO_URL_METRICS_POST_TYPE
-						)
-					);
-				}
-
-				return $is_valid;
-			}
+			array_map(
+				static function ( $url_metric_data ) use ( $trigger_error ) {
+					if ( ! is_array( $url_metric_data ) ) {
+						return null;
+					}
+
+					try {
+						// TODO: This is re-validating the data which has been stored in the post type. This ensures it remains valid, but is it overkill?
```

Comment:
My only hesitation is that this additional validation adds latency, so it negatively impacts TTFB. However, I haven't profiled the JSON Schema validation logic, so this may not be a big deal.